library/std/src/sys/hermit/condvar.rs

```diff
@@ -55,8 +55,18 @@ impl Condvar {
         mutex.lock();
     }
 
-    pub unsafe fn wait_timeout(&self, _mutex: &Mutex, _dur: Duration) -> bool {
-        panic!("wait_timeout not supported on hermit");
+    pub unsafe fn wait_timeout(&self, mutex: &Mutex, dur: Duration) -> bool {
+        if dur.is_zero() {
+            return false;
```

Comment:
@mkroening Why is this always `false`? The semaphore could be available?

Reply:
Addressed in https://github.com/hermitcore/rust/pull/5